### One Tailed (Right Tailed) - One-Sample Z-Test

**1 - What is it?**

A. `Right One-Tailed One-Sample Z-Test` check the means of sample is significantly higher from the population mean (when population standard devidation is known and sample size is large, usually n>=30).

`Think of i like` Is my small group's average weight `really higher (significantly)` from the entire country's average weight, or it just random chance.

**2 - Real-Life Example : Right-Tailed One-Sample Z-Test**

`Scenario:` Your marketing team says : `Our new landing page increase average session time beyond 5 minutes`

You want to test if its really greater than 5 minutes, not just different - So it's a one-tailed test (specifically, right-tailed)

**3 - Hypotheses**

- H0 (Null Hypothesis) : Population mean (meau) = 5 (no improement).
- H1 (Alternate Hypothesis) : Population mean > Sample mean (mean time increased).

This is a `right-tailed` test because we're checking the `greater than`.

**4 - Formula**

Z = (Sample mean-Population mean) / (Populatin std / Sqrt(Sample size))

Z = (x bar - μ) / (σ/sqrt(n))

**Where: / Parameters**
- Population mean (μ) = 5 min
- Population std (σ) = 1 min
- Sample mean (x bar) = 5.5 min
- Sample size (n) = 50
- Significance level (alpha) = 0.05
- Confidence Interval = 0.95

**4 - Formula**

Z = (Sample mean - Population mean) / (Population standard deviation / Sqrt(Sample size))

Z = (x bar - μ) / (σ/sqrt(n))

**Where: / Parameters**
- population mean(μ) = 5 min
- populatin standard devidation(σ) = 1 min
- sample mean(x bar) = 5.5 min
- sample size(n) = 50
- signficance level (alpha) = 0.05
- confidence interval = 0.95

**5 - Python Implementation (Step-by-Step)**

In [12]:
import numpy as np
from scipy import stats

# Step 1 : Define parameters
population_mean = 5 # average time on site (in minute)
population_std = 1 # population standard deviation (spread from mean)
sample_mean = 5.5 # sample mean
sample_size = 50 # number of visitors in sample
alpha = 0.05 # significance level

# Step 2 : Calculate the Standard Error (SE)
standard_error = population_std / np.sqrt(sample_size)
print(f"Standard Error: {round(standard_error,4)}")

# Step 3 : Z Statistics
z_stat = (sample_mean - population_mean) / standard_error
print(f"Z-Statistics: {round(z_stat,4)}")

# Step 4 : Calculate Right One-Tailed One-Sample Z-Test P-Value
p_value = 1 - (stats.norm.cdf(round(z_stat,4)))
print(f"P Value: {round(p_value,4)}")

# Step 5 : Decision
if p_value < alpha:
    print("Rejecting H0: Landing page significantly increased session time.")
else:
    print("Fail to Reject H1: No signficant increase in session time.")

Standard Error: 0.1414
Z-Statistics: 3.5355
P Value: 0.0002
Rejecting H0: Landing page significantly increased session time.


**6 - Business Interpretation**

- Z = 3.53 -> Sample mean is 3.5 standard errors above the population mean.
- p = 0.0002 -> There's only 0.02% chance that this result happened by luck.
- So, here new page `really did` increase time - not by random luck.